hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -91,6 +94,13 @@ public HoodieBaseFileGroupRecordBuffer(HoodieReaderContext<T> readerContext,
     this.partitionPathFieldOpt = partitionPathFieldOpt;
     this.recordMergeMode = getRecordMergeMode(props);
     this.recordMerger = recordMerger;
+    this.orderingFieldName = Option.ofNullable(ConfigUtils.getOrderingField(props)).orElseGet(() -> hoodieTableMetaClient.getTableConfig().getPreCombineField());
+    if (this.orderingFieldName == null) {
+      this.orderingFieldType = Schema.Type.INT;
+    } else {
+      this.orderingFieldType = HoodieAvroUtils.getActualSchemaFromUnion(readerSchema.getField(orderingFieldName).schema(), null).getType();
```

Comment:
Is it possible to have complex data structure, like map, or record?

Reply:
no